File: C#/PlusOne.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        
        List<int> Digits = digits.ToList();
        
        Digits[Digits.Count - 1]++;
        
        
        if (Digits[Digits.Count - 1] == 10)
        {
            int Index = Digits.Count - 1;
            while (Digits[Index] == 10 && Index > 0)
            {
                Digits[Index] = 0;
                Digits[Index - 1] += 1;
                Index--;
            }
            
            if (Digits[0] == 10)
            {
                Digits[0] = 0;
                Digits.Insert(0, 1);
            }
        }
        
        return Digits.ToArray();
    }
}

File: Python/plusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:

        digit = 0

        for val in digits:
            digit *= 10
            digit += val

        digit += 1

        output = []

        while digit > 0:
            output.append(digit % 10)
            digit = digit // 10

        output.reverse()
        return output